src/core/sort.js

```diff
@@ -1,3 +1,7 @@
+/**
+ * @typedef {import('../framework/components/component').Component} Component
```

Comment:
we should not be importing anything from framework (which is the top level layer in the engine) into core (which is the bottom level).

I'd just move those two functions to where we need them, as at this point they're no longer general.